onmt/dynamic/opts.py

```diff
@@ -24,6 +22,23 @@ def _dynamic_corpus_opts(parser):
               help="Default transform pipeline applying to data."
                    "Can be specified in each corpus of data to override.")
 
+    group.add('-save_data', '--save_data', required=build_vocab_only,
+              help='Output directory for data saving path.')
+    group.add(
+        '-n_sample', '--n_sample',
+        type=int, default=(5000 if build_vocab_only else 0),
+        help=("Build vocab using " if build_vocab_only else "Stop after save ")
+        + "this number of transformed samples/corpus. Can be [-1, 0, N>0]. "
+        "Set to -1 to go full corpus, 0 to skip.")
+
+    if not build_vocab_only:
+        group.add('-dump_fields', '--dump_fields', action='store_true',
+                  help="Dump fields `*.vocab.pt` to disk."
+                  " -save_data should be set as saving prefix.")
```

Comment:
Wouldn't it be more straightforward if `-dump_fields` was the path where we want to save it, not only a boolean flag?
(same for `-dump_transforms`)

Reply:
not sure about embeddings though

Reply:
Use a single `save_data` to control the saving path for `fields/transforms/samples/pretrained embeddings` seems cleaner to me rather than give one path for each which needs copy/paste in most cases.

Reply:
I think both approaches are valid. But alright let's keep `save_data` for now.